### ISSUE TITLE
BUG: fs.rename gives incorrect error message on error

### ISSUE THREAD

**Issue by isaacs** [joyent/node#685]
Try this:

    mkdir -p ./path/that/exists/
    touch ./path/that/exists/file

test.js

    (function () {
      "use strict";

      var fs = require("fs");

      fs.rename("path/that/exists/file", "path/that/doesntexist/file", function (err) {
        console.log(err);
      });
    }());

node test.js

    { message: 'ENOENT, No such file or directory \'path/that/exists/file\''
    , stack: [Getter/Setter]
    , errno: 2
    , path: 'path/that/exists/file'
    }

expected:

    { message: 'ENOENT, No such file or directory \'path/that/doesntexists/file\''
    , stack: [Getter/Setter]
    , errno: 2
    , path: 'path/that/exists/file'
    }

AJ ONeal

**Comment by robertkowalski** [comment]
@piscisaureus is this still an issue in io.js?

**Comment by mhdawson** [comment]
Confirmed still  exists on v0.12 for ubuntu

**Comment by mhdawson** [comment]
Confirmed issues exists in io.js as well

**Comment by piscisaureus** [comment]
Libuv (at least on windows, don't know about unix) can't tell whether an error is related to the 'source' or 'target' path. Therefore, the only option we have is to display both paths.

io.js does that:

```
C:\Users\Bert Belder>iojs
> fs.renameSync('foo', 'bar')
Error: ENOENT: no such file or directory, rename 'C:\Users\Bert Belder\foo' -> '
C:\Users\Bert Belder\bar'
    at Error (native)
    at Object.fs.renameSync (fs.js:685:18)
    at repl:1:4
    at REPLServer.defaultEval (repl.js:155:27)
    at bound (domain.js:254:14)
    at REPLServer.runBound [as eval] (domain.js:267:12)
    at REPLServer.<anonymous> (repl.js:309:12)
    at emitOne (events.js:77:13)
    at REPLServer.emit (events.js:166:7)
    at REPLServer.Interface._onLine (readline.js:208:10)
>
```

(@mhansen - you reported that this is still an issue in io.js, but in fact this was fixed late January in https://github.com/nodejs/io.js/commit/bc2c85ceef7ac034830e4a4357d0aef69cd6e386. I suspect that you may have accidentally tested node or a really old io.js version)

**Comment by jasnell** [comment]
Ok, so the question that remains is: do we want to backport this fix into v0.12 or joyent/node master or pick it up when we move to the converged stream?

**Comment by mhdawson** [comment]
I think I ran on a recent enough io.js but when we reviewed the message the initial thought was that it still was not clear enough.  Given your explanation about not being able to tell which path is the problem I can see its the best we can do.

Since its a P3 it should be able to wait until the converged stream.

**Comment by jasnell** [comment]
Agreed on waiting for the converged stream.

**Comment by jasnell** [comment]
at this point, I'm inclined to close this here. If someone wishes to backport the io.js fix to v0.12 then so be it, but the priority is low.
